aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -56,7 +56,7 @@ __global__ void max_pool_forward_nchw(const int nthreads, const scalar_t* bottom
       wstart += dilation_w;
     accscalar_t maxval = at::numeric_limits<accscalar_t>::lower_bound(); // -Infinity
     int maxidx = hstart * width + wstart;
-    const scalar_t* btm_data = bottom_data + (n * channels + c) * height * width;
+    const scalar_t* btm_data = bottom_data + static_cast<ptrdiff_t>(n * channels + c) * height * width;
```

Comment:
better  cast `height` and `width` and `channels` to `ptrdiff_t`, otherwise `height * width` can overflow, or `n*channels + c` can overflow

Reply:
Should we have a generic c10::scalar_with_overflow that will trigger an error if there is an overflow? Or use some kind of BigInteger library (GMP springs to mind)

Reply:
We can't have real int64 indexing overflows on the device, it exceeds currently available memory (and memory likely to be available in the next decade) by several orders of magnitude.